### ISSUE TITLE
Raspberry Pi 32bit Segfault - OpenCV

### ISSUE THREAD

**Issue by peardox** [bytedeco/javacpp-presets#805]
Compilation test using Oracle JDK 8

[debug.opencv.log.gz](https://github.com/bytedeco/javacpp-presets/files/3729616/debug.opencv.log.gz)

This appears to be the error in question...

opencv/target/native/org/bytedeco/opencv/linux-armhf/jniopencv_core.cpp:124812:1: internal compiler error: Segmentation fault

A nice specific error this time with -X -e

I'll go give 11 a try

**Comment by saudet** [comment]
That just looks like it's running out of memory. Make sure that's not happening by monitoring memory usage with `top`.

**Comment by saudet** [comment]
GCC is the issue, not Maven. I'm not aware of any such switch for GCC. Try to add more swap space.

**Comment by peardox** [comment]
I'm using 8G Swap on a 4G 32 bit Pi.

A potential problem is the 4G address space on 32 but OSes as it gets much further on my 64 bit Nano.

I'll mess around a bit and see what I can come  up with...

**Comment by saudet** [comment]
Right, it won't be able to get much higher than 2 or 3 GB... Disabling things like compiler optimizations usually reduces memory usage.

**Comment by peardox** [comment]
I just gave it another do with swappiness set to 90 - still failed

Next idea - try compiling it on 64 bit with a -m32 switch - it compiles fine on arm64, If this works then all the cppbuild.sh scripts are going to need thee -32 - worry about that if test works

While it's not an ideal solution it's an acceptable one IMHO as it only requires a 64 bit OS to build (I hope)

**Comment by peardox** [comment]
Something nice - gcc-multilib-arm-linux-gnueabihf includes arm-linux-gnueabihf-gcc on 64-bit so no changes required to cppbuild.sh, it appears

JavaCPP compiled correctly

**Comment by peardox** [comment]
Oh well - too much to hope for I guess. It's trying to compile some 64 bit ASM

Invoked with

`mvn install -Djavacpp.platform=linux-armhf -Dmaven.javadoc.skip=true -DVERBOSE=1`

`make[1]: warning: -jN forced in submake: disabling jobserver mode.
make[1]: *** No rule to make target '../kernel/arm64/amin.S', needed by 'samin_k.o'.  Stop.
make[1]: *** Waiting for unfinished jobs....
../kernel/arm64/amax.S: Assembler messages:
../kernel/arm64/amax.S:162: Error: ARM register expected -- `cmp x0,xzr'
../kernel/arm64/amax.S:164: Error: ARM register expected -- `cmp x2,xzr'
../kernel/arm64/amax.S:167: Error: ARM register expected -- `cmp x2,#1'
../kernel/arm64/amax.S:172: Error: ARM register expected -- `asr x5,x0,#2'
../kernel/arm64/amax.S:173: Error: ARM register expected -- `cmp x5,xzr'
../kernel/arm64/amax.S:176: Error: bad instruction `ld1 {v0.4s},[x1],#16'
../kernel/arm64/amax.S:176: Error: bad instruction `fabs v0.4s,v0.4s'
../kernel/arm64/amax.S:176: Error: bad instruction `fmaxv s0,v0.4s'
../kernel/arm64/amax.S:177: Error: ARM register expected -- `subs x5,x5,#1'
../kernel/arm64/amax.S:182: Error: bad instruction `ld1 {v1.4s},[x1],#16'
../kernel/arm64/amax.S:182: Error: bad instruction `fabs v1.4s,v1.4s'
../kernel/arm64/amax.S:182: Error: bad instruction `fmaxv s1,v1.4s'
../kernel/arm64/amax.S:182: Error: bad instruction `fcmp s0,s1'
../kernel/arm64/amax.S:182: Error: bad instruction `fcsel s0,s0,s1,ge'
../kernel/arm64/amax.S:184: Error: ARM register expected -- `subs x5,x5,#1'
../kernel/arm64/amax.S:189: Error: ARM register expected -- `ands x5,x0,#3'
../kernel/arm64/amax.S:194: Error: ARM register expected -- `ldr s1,[x1],#4'
../kernel/arm64/amax.S:194: Error: bad instruction `fabs s1,s1'
../kernel/arm64/amax.S:194: Error: bad instruction `fcmp s0,s1'
../kernel/arm64/amax.S:194: Error: bad instruction `fcsel s0,s0,s1,ge'
../kernel/arm64/amax.S:196: Error: ARM register expected -- `subs x5,x5,#1'
../kernel/arm64/amax.S:199: Error: bad instruction `ret'
../kernel/arm64/amax.S:203: Error: ARM register expected -- `ldr s0,[x1],#4'
../kernel/arm64/amax.S:203: Error: bad instruction `fabs s0,s0'
../kernel/arm64/amax.S:204: Error: ARM register expected -- `subs x0,x0,#1'
../kernel/arm64/amax.S:209: Error: ARM register expected -- `lsl x2,x2,#2'
../kernel/arm64/amax.S:209: Error: bad instruction `ld1 {v0.s}[0],[x1],x2'
../kernel/arm64/amax.S:209: Error: bad instruction `fabs s0,s0'
../kernel/arm64/amax.S:211: Error: ARM register expected -- `subs x0,x0,#1'
../kernel/arm64/amax.S:214: Error: ARM register expected -- `asr x5,x0,#2'
../kernel/arm64/amax.S:215: Error: ARM register expected -- `cmp x5,xzr'
../kernel/arm64/amax.S:220: Error: bad instruction `ld1 {v1.s}[0],[x1],x2'
../kernel/arm64/amax.S:220: Error: bad instruction `fabs s1,s1'
../kernel/arm64/amax.S:220: Error: bad instruction `fcmp s0,s1'
../kernel/arm64/amax.S:220: Error: bad instruction `fcsel s0,s0,s1,ge'
../kernel/arm64/amax.S:221: Error: bad instruction `ld1 {v1.s}[0],[x1],x2'
../kernel/arm64/amax.S:221: Error: bad instruction `fabs s1,s1'
../kernel/arm64/amax.S:221: Error: bad instruction `fcmp s0,s1'
../kernel/arm64/amax.S:221: Error: bad instruction `fcsel s0,s0,s1,ge'
../kernel/arm64/amax.S:222: Error: bad instruction `ld1 {v1.s}[0],[x1],x2'
../kernel/arm64/amax.S:222: Error: bad instruction `fabs s1,s1'
../kernel/arm64/amax.S:222: Error: bad instruction `fcmp s0,s1'
../kernel/arm64/amax.S:222: Error: bad instruction `fcsel s0,s0,s1,ge'
../kernel/arm64/amax.S:223: Error: bad instruction `ld1 {v1.s}[0],[x1],x2'
../kernel/arm64/amax.S:223: Error: bad instruction `fabs s1,s1'
../kernel/arm64/amax.S:223: Error: bad instruction `fcmp s0,s1'
../kernel/arm64/amax.S:223: Error: bad instruction `fcsel s0,s0,s1,ge'
../kernel/arm64/amax.S:225: Error: ARM register expected -- `subs x5,x5,#1'
../kernel/arm64/amax.S:230: Error: ARM register expected -- `ands x5,x0,#3'
../kernel/arm64/amax.S:235: Error: bad instruction `ld1 {v1.s}[0],[x1],x2'
../kernel/arm64/amax.S:235: Error: bad instruction `fabs s1,s1'
../kernel/arm64/amax.S:235: Error: bad instruction `fcmp s0,s1'
../kernel/arm64/amax.S:235: Error: bad instruction `fcsel s0,s0,s1,ge'
../kernel/arm64/amax.S:237: Error: ARM register expected -- `subs x5,x5,#1'
../kernel/arm64/amax.S:242: Error: bad instruction `ret'
../kernel/arm64/amax.S:246: Error: bad instruction `fmov s0,wzr'
../kernel/arm64/amax.S:247: Error: bad instruction `ret'
Makefile.L1:525: recipe for target 'samax_k.o' failed
make[1]: *** [samax_k.o] Error 1
Makefile:149: recipe for target 'libs' failed
make: *** [libs] Error 1
`
